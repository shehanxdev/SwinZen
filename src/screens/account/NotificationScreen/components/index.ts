export * from './NotificationCard';
export * from './SectionHeader';
