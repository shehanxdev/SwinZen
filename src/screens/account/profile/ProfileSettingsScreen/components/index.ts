export * from './ProfileImageUpload';
export * from './ProfileInfoCard';
