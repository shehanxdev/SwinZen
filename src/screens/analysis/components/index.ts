export * from './BaseAnalysisScreen';
export * from './TipsBottomCard';
