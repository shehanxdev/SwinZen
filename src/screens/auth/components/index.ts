export * from './BaseAuthScreen';
export * from './OTPInput';
export * from './PasswordField';
