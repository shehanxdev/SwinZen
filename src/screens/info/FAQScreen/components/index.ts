export * from './FAQSectionContent';
export * from './FAQSectionHeader';
