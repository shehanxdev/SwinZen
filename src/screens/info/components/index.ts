export * from './BaseScreen';
export * from './FAQSectionHeader';
export * from './FAQSectionContent';
