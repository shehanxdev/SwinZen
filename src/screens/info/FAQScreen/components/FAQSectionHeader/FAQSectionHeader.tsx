import React from 'react';
import { View } from 'react-native';

import { DownArrowIcon, Text, UpArrowIcon } from '@sz/components';
import { tw } from '@sz/config';
import { Color, TextAlignment, TextVariant } from '@sz/constants';
import { FaqSection } from '@sz/models';

interface FAQSectionHeaderPropTypes {
  content: FaqSection;
  index: number;
  isActive: boolean;
}

export function FAQSectionHeader({ content, isActive }: FAQSectionHeaderPropTypes) {
  return (
    <View style={tw`flex-row justify-between`} testID="FAQSectionHeaderTestID">
      <View style={tw`max-w-[90%]`}>
        <Text variant={TextVariant.Body2SemiBold} color={Color.Neutral.Sz100} textAlign={TextAlignment.Left}>
          {content.question}
        </Text>
      </View>
      {isActive ? <DownArrowIcon /> : <UpArrowIcon />}
    </View>
  );
}
