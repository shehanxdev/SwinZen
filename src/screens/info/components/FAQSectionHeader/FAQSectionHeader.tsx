import React from 'react';
import { View } from 'react-native';

import { DownArrowIcon, Text, UpArrowIcon } from '@sz/components';
import { tw } from '@sz/config';
import { Color, TextAlignment, TextVariant } from '@sz/constants';

export function FAQSectionHeader(section, _, isActive) {
  return (
    <View style={tw`flex-row justify-between`}>
      <View style={tw`max-w-[90%]`}>
        <Text variant={TextVariant.Body2SemiBold} color={Color.Neutral.Sz100} textAlign={TextAlignment.Left}>
          {section.title}
        </Text>
      </View>
      {isActive ? <DownArrowIcon /> : <UpArrowIcon />}
    </View>
  );
}
