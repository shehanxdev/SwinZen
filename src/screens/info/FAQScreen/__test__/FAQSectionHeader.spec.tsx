import { render } from '@testing-library/react-native';

import { FAQSectionHeader } from '../components/FAQSectionHeader';

describe('FAQ section header', () => {
  const testID = 'FAQSectionHeaderTestID';
  const dummyFAQ = [
    {
      title: 'Lorem ipsum dolor sit amet, consectetur',
      content:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean non dolor purus. Nunc semper sapien congue tortor',
    },
  ];

  const getRenderedComponent = () => render(<FAQSectionHeader content={dummyFAQ[0]} index={1} isActive={true} />);

  it(`should render FAQ Section content correctly`, () => {
    const renderer = getRenderedComponent();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it('should find the FAQ Screen via testID', () => {
    const { getByTestId } = getRenderedComponent();
    const foundScreen = getByTestId(testID);
    expect(foundScreen).toBeTruthy();
  });
});
