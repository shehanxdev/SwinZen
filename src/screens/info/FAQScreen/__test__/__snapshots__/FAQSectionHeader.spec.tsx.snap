// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ section header should render FAQ Section content correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
  testID="FAQSectionHeaderTestID"
>
  <View
    style={
      {
        "maxWidth": "90%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#F8F9FA",
            "lineHeight": 24,
          },
          {
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 16,
            "fontWeight": "600",
            "textAlign": "left",
          },
        ]
      }
    >
      Lorem ipsum dolor sit amet, consectetur
    </Text>
  </View>
  <SvgMock
    fill="#FFFFFF"
    height={24}
    width={24}
  />
</View>
`;
